packages/fast-components/src/anchored-region/anchored-region.ts

```diff
@@ -0,0 +1,978 @@
+                        ? horizontalOptions[0]
+                        : horizontalOptions[1];
+            }
+        }
+
+        if (this.verticalPositioningMode !== "uncontrolled") {
+            const verticalOptions: AnchoredRegionVerticalPositionLabel[] = this.getVerticalPositioningOptions();
+            if (this.verticalDefaultPosition !== "unset") {
+                switch (this.verticalDefaultPosition) {
+                    case "top":
+                        desiredVerticalPosition = this.verticalInset
+                            ? AnchoredRegionVerticalPositionLabel.insetTop
+                            : AnchoredRegionVerticalPositionLabel.top;
+                        break;
+
+                    case "bottom":
+                        desiredVerticalPosition = this.verticalInset
+                            ? AnchoredRegionVerticalPositionLabel.insetBottom
+                            : AnchoredRegionVerticalPositionLabel.bottom;
+                        break;
+                }
+            }
+
+            const verticalThreshold: number =
+                this.verticalThreshold !== undefined
+                    ? Number(this.verticalThreshold)
+                    : this.positionerDimension.height;
+
+            if (
+                desiredVerticalPosition ===
+                    AnchoredRegionVerticalPositionLabel.undefined ||
+                (!(this.verticalPositioningMode === "locktodefault") &&
+                    this.getAvailableHeight(desiredVerticalPosition) < verticalThreshold)
+            ) {
+                desiredVerticalPosition =
+                    this.getAvailableHeight(verticalOptions[0]) >
+                    this.getAvailableHeight(verticalOptions[1])
+                        ? verticalOptions[0]
+                        : verticalOptions[1];
+            }
+        }
+
+        const nextPositionerDimension: Dimension = this.getNextRegionDimension(
+            desiredHorizontalPosition,
+            desiredVerticalPosition
+        );
+
+        const positionChanged: boolean = !(
+            this.currentHorizontalPosition === desiredHorizontalPosition &&
+            this.currentVerticalPosition === desiredVerticalPosition
+        );
+
+        this.setHorizontalPosition(desiredHorizontalPosition, nextPositionerDimension);
+        this.setVerticalPosition(desiredVerticalPosition, nextPositionerDimension);
+        this.transformOrigin = `${this.yTransformOrigin} ${this.xTransformOrigin}`;
+
+        this.updateRegionStyle();
+
+        this.initialLayoutComplete = true;
+
+        if (positionChanged) {
+            this.positionChanged();
+        }
+    };
+
+    /**
+     *  Updates the style string applied to the region element as well as the css classes attached
+     *  to the root element
+     */
+    private updateRegionStyle = (): void => {
+        this.classList.toggle("horizontalInset", this.horizontalInset);
```

Comment:
cool - I didn't know `classList` had a toggle method!

Reply:
Yes, it's handy.